File: Planner/app/src/main/java/eugene/example/helloapplication/MainPage/MainActivity.java
package eugene.example.helloapplication.MainPage;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import eugene.example.helloapplication.Menu.About_activity;
import eugene.example.helloapplication.Plans.Plans;
import eugene.example.helloapplication.R;
import eugene.example.helloapplication.Timer.Timer;
import eugene.example.helloapplication.Menu.main_Information_about_program;
import timber.log.Timber;

public class MainActivity extends AppCompatActivity {
    public Timer timer = new Timer();
    public static String EXTR;
    public final static String K_REV = "k_rev";
    public int revenue=0;
    static {
        EXTR = "EXTRA_MESSAGE";
    }



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_page);
        Timber.i("---------------------------onCreate Called.------------------------");
        timer.startTimerTotal();
        if (savedInstanceState!=null)
        {
            revenue = savedInstanceState.getInt(K_REV, 1);
        }
        MainActivity.EXTR = EXTR;
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(K_REV,timer.getSecondsCountFocused());
        Timber.i("onSaveInstanceState Called");
    }
    @Override
    public void onStart()
    {
        super.onStart();
        Timber.i("----------onStart Called.----------");
    }
    @Override
    public void onPause()
    {
        super.onPause();
        timer.stopTimerFocused();
        Timber.i("----------onPause Called.----------");
    }
    @Override
    public void onDestroy()
    {
        super.onDestroy();
        Timber.i("----------onDestroy Called.----------");
        timer.stopTimerTotal();
    }
    @Override
    public void onResume()
    {
        super.onResume();
        timer.startTimerFocused();
        Timber.i("----------onResume Called.----------");
    }
    @Override
    public void onStop()
    {
        super.onStop();
        Timber.i("----------onStop Called.----------");
    }
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        TextView headerView = (TextView) findViewById(R.id.header);
        switch(id){
            case R.id.open_activity :
                Intent intent = new Intent(this, Plans.class);
                Timber.i("---------------------------Активирован пункт меню открытия распорядка дня------------------------");
                startActivity(intent);
                return true;
            case R.id.about:
                intent = new Intent(this, About_activity.class);
                Timber.i("---------------------------Активирован пункт меню про разработчика------------------------");
                startActivity(intent);
                return true;
            case R.id.Rules:
                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT, "I`m 20 years old, living in Chernivtsi, Ukraine. For \n" +
                        "whole my life i was always interested in Computer \n" +
                        "Science and tried to understand how to use this \n" +
                        "passion in future. When i had to decide which \n" +
                        "profession would accomplish earning money and \n" +
                        "doing what i love most i found out that Department \n" +
                        "of Computer Science in Institute of Physical, \n" +
                        "Technical and Computer Sciences would be the best \n" +
                        "decision. I always trying to learn new information \n" +
                        "from everywhere: from university, from different \n" +
                        "internet sources etc. I would like to work in \n" +
                        "team because of my sociable skills and leader \n" +
                        "qualities. I can perceive a new information and use it \n" +
                        "as quick as it possible.\n" +
                        "VIKTOR\n" +
                        "KOROLENKO");
                sendIntent.setType("text/*");
                Intent shareIntent = Intent.createChooser(sendIntent, null);
                Timber.i("---------------------------Активирован пункт меню для отправки------------------------");
                startActivity(shareIntent);
                return true;
            case R.id.information:
                intent = new Intent(this, main_Information_about_program.class);
                Timber.i("---------------------------Активирован пункт меню про программу------------------------");
                startActivity(intent);
                return true;
            case R.id.exit:
                Timber.i("---------------------------выход из программы------------------------");
                onDestroy();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
    public void sendMessage(View view)
    {
            // действия, совершаемые после нажатия на кнопку
            // Создаем объект Intent для вызова новой Activity
            Intent intent = new Intent(this, Plans.class);
            // Получаем текстовое поле в текущей Activity
            EditText editText = (EditText) findViewById(R.id.edit_message);
            // Получае текст данного текстового поля
            String message = editText.getText().toString();
            // Добавляем с помощью свойства putExtra объект - первый параметр - ключ,
            // второй параметр - значение этого объекта
            intent.putExtra(EXTR, message);
            // запуск activity
            startActivity(intent);
            Timber.plant();
            Timber.i("------------------sendmessage method used--------------------");
    }
    public void Exit(View view)
    {
        onDestroy();
    }

}

File: app/src/main/java/viktor/braus/kplanner/mainPage/MainActivity.kt
package viktor.braus.kplanner.mainPage

import android.annotation.SuppressLint
import android.app.TimePickerDialog
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import timber.log.Timber
import viktor.braus.kplanner.R
import viktor.braus.kplanner.entity.PlansDAO
import viktor.braus.kplanner.entity.PlansDatabase
import viktor.braus.kplanner.menu.factoryMethod.About_Fragment
import viktor.braus.kplanner.menu.viewModel.Information_fragment
import viktor.braus.kplanner.plans.listOfPlans.ListFragment
import viktor.braus.kplanner.plans.plansCreating.PlansFactory
import viktor.braus.kplanner.plans.plansCreating.PlansFragment
import viktor.braus.kplanner.plans.plansCreating.PlansViewModel
import viktor.braus.kplanner.timer.TTimer
import java.text.SimpleDateFormat
import java.util.*


//"Перейшов на мову Kotlin, 5 Лабораторна робота"
class MainActivity : AppCompatActivity() {
    var d: String? = null
    var fragment = MainFragment()
    var TTimer: TTimer = TTimer()
    var time: String? = null
    companion object {
        var K_REV: String? = null
        var S: String? = null
    }
        var revenue = 0
    init
    {
        K_REV = "k_rev"
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_page)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.main_container, fragment)
            .addToBackStack(null)
            .commit()
        Timber.i("---------------------------onCreate Called.------------------------")
        TTimer.startTimerTotal()
        if (savedInstanceState != null) {
            revenue = savedInstanceState.getInt(K_REV, 1)
        }

    }
    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
    }
    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt(K_REV, TTimer.secondsCountFocused)
        Timber.i("onSaveInstanceState Called")
    }
    override fun onStart() {
        super.onStart()
        Timber.i("----------onStart Called.----------")
    }

    override fun onPause() {
        super.onPause()
        TTimer.stopTimerFocused()
        Timber.i("----------onPause Called.----------")
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.i("----------onDestroy Called.----------")
        TTimer.stopTimerTotal()
    }
    override fun onResume() {
        super.onResume()
        TTimer.startTimerFocused()
        Timber.i("----------onResume Called.----------")
    }

    override fun onStop() {
        super.onStop()
        Timber.i("----------onStop Called.----------")
    }
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        when (id) {
            R.id.open_activity -> {
                Timber.i("---------------------------Активирован пункт меню открытия распорядка дня------------------------")
                var fragment = ListFragment()
                supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
                return true
            }
            R.id.about -> {
                var fragment = About_Fragment()
                supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
                Timber.i("---------------------------Активирован пункт меню про разработчика------------------------")
                return true
            }
            R.id.Rules -> {
                val sendIntent = Intent()
                sendIntent.action = Intent.ACTION_SEND
                sendIntent.putExtra(
                    Intent.EXTRA_TEXT,
                    """
                I`m 20 years old, living in Chernivtsi, Ukraine. For 
                whole my life i was always interested in Computer 
                Science and tried to understand how to use this 
                passion in future. When i had to decide which 
                profession would accomplish earning money and 
                doing what i love most i found out that Department 
                of Computer Science in Institute of Physical, 
                Technical and Computer Sciences would be the best 
                decision. I always trying to learn new information 
                from everywhere: from university, from different 
                internet sources etc. I would like to work in 
                team because of my sociable skills and leader 
                qualities. I can perceive a new information and use it 
                as quick as it possible.
                VIKTOR
                KOROLENKO
                """.trimIndent()
                )
                sendIntent.type = "text/*"
                val shareIntent = Intent.createChooser(sendIntent, null)
                Timber.i("---------------------------Активирован пункт меню для отправки------------------------")
                startActivity(shareIntent)
                return true
            }
            R.id.information -> {
                var fragment = Information_fragment()
                supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
                Timber.i("---------------------------Активирован пункт меню про программу------------------------")
                startActivity(intent)
                return true
            }
            R.id.exit -> {
                Timber.i("---------------------------выход из программы------------------------")
                onDestroy()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    fun sendMessage(view: View) {
        val editText = findViewById<View>(R.id.edit_message) as EditText
        var message = editText.text.toString()
        Timber.plant()
        Timber.i("------------------sendmessage method used--------------------")
        S = message
        Timber.i("Имя пользователя, которое было отправлено: $message")
        var fragment = ListFragment()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.main_container, fragment)
            .addToBackStack(null)
            .commit()
    }
    fun Exit(view: View)
    {
        finishAffinity()
    }
    fun BackWard(view: View?)
    {
        Timber.i("---------------------------Возвращение на главный экран------------------------")
        var fragment : MainFragment  = MainFragment()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.main_container, fragment)
            .addToBackStack(null)
            .commit()
    }
    fun returning(v:View) {
        Timber.i("---------------------------Возвращение на страницу распорядка------------------------")
        var fragment = ListFragment()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.main_container, fragment)
            .addToBackStack(null)
            .commit()
    }

    @SuppressLint("SetTextI18n")
    fun goNext(view: View?) {

    }
    fun Planning(view: View?) {
        Timber.i("---------------------------Активировано действие перехода на создание плана------------------------")
        var fragment = PlansFragment()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.main_container, fragment)
            .addToBackStack(null)
            .commit()

    }
//    @SuppressLint("SimpleDateFormat")
//    fun time1(view: View?)
//    {
//        val cal = Calendar.getInstance()
//        var con = PlansFragment().activity
//        val dataSource = PlansDatabase.getInstance(application).plansDAO
//        TimePickerDialog(
//            con,
//            PlansViewModel(application, dataSource).timeSetListener, cal.get(Calendar.HOUR_OF_DAY
//            ), cal.get(Calendar.MINUTE), true
//        ).show()
//        PlansViewModel(application, dataSource).time1()
//    }
    @SuppressLint("SimpleDateFormat", "SetTextI18n")
    fun setTime(v: View)
    {
        var t: TextView = findViewById(R.id.timeEvent)
        var t1: TextView = findViewById(R.id.startTimeEvent)
        var t2: TextView = findViewById(R.id.endTimeEvent)
        val id = v.id
        if(id == R.id.timeButton) {
            val cal = Calendar.getInstance()
            var timeSetListener = TimePickerDialog.OnTimeSetListener { timePicker, hour, minute ->
                cal.set(Calendar.HOUR_OF_DAY, hour)
                cal.set(Calendar.MINUTE, minute)
                var t: TextView = findViewById(R.id.timeEvent)
                t.text = SimpleDateFormat().format(cal.time)
                d=t.text.toString()
                Timber.i("nnnnnnnnnnn${d}")
            }
            TimePickerDialog(
                this@MainActivity, timeSetListener,
                cal.get(Calendar.HOUR_OF_DAY),
                cal.get(Calendar.MINUTE), true
            ).show()
        }
        if(id==R.id.timeStartButton)
        {
            val cal = Calendar.getInstance()
            var timeSetListener = TimePickerDialog.OnTimeSetListener { timePicker, hour, minute ->
                cal.set(Calendar.HOUR_OF_DAY, hour)
                cal.set(Calendar.MINUTE, minute)
                t1.text = SimpleDateFormat().format(cal.time)
                d=t1.text.toString()
            }
            TimePickerDialog(
                this@MainActivity, timeSetListener,
                cal.get(Calendar.HOUR_OF_DAY),
                cal.get(Calendar.MINUTE), true
            ).show()
        }
        if(id==R.id.timeEndButton)
        {
            val cal = Calendar.getInstance()
            var timeSetListener = TimePickerDialog.OnTimeSetListener { timePicker, hour, minute ->
                cal.set(Calendar.HOUR_OF_DAY, hour)
                cal.set(Calendar.MINUTE, minute)
                t2.text = SimpleDateFormat().format(cal.time)
                d=t2.text.toString()
            }
            TimePickerDialog(
                this@MainActivity, timeSetListener,
                cal.get(Calendar.HOUR_OF_DAY),
                cal.get(Calendar.MINUTE), true
            ).show()
        }
    }
}